common/models/user.js

```diff
@@ -50,6 +54,23 @@ module.exports = function(User) {
   // see:
   // https://github.com/strongloop/loopback/issues/1137#issuecomment-109200135
   delete User.validations.email;
+  // Updated workaround for email validation
+  // see:
+  // https://github.com/strongloop/loopback/issues/1137#issuecomment-271437926
+  User.validate('email', function(err) {
+    if (this.email && !isEmail(this.email)) {
+      // Here err is a callback
+      // See loopback api docs loopback-datasource-juggler/#validatable-validate
+      debug(err());
```

Comment:
Okay, I re-visited this entire thing once again. And turns out with your last change to the email this block:

https://github.com/raisedadead/freeCodeCamp/blob/3f79222a25ee35b013646cd513f4eddc9dc0220f/common/models/user.js#L52-L68

is never executed, and the handler seems to catch it all correctly. 

I'll confirm this once again, but basically we can get rid of the block.